consul/consul.go

```diff
@@ -0,0 +1,96 @@
+package consul
+
+import (
+	"github.com/hashicorp/consul/agent/consul"
+	"github.com/hashicorp/consul/api"
+)
+
+// Client provides a client to functions of Consul API
+// for one serivce
+type Client struct {
+	clientAPI *api.Client
+	srvID     string
+}
+
+// NewDefaultClient is the most default constructor for Consul agent for one serivce.
+// This function gets the local IP address and register service for a local agent with a given name.
+// It ignores errors if Consul agent is unreachable.
+// srvName sets for ID, Name and CheckID of service.
+func NewDefaultClient(srvName string, svcPort int, checkTTL string) (*Client, error) {
+	var err error
+
+	client := &Client{srvID: srvName}
+
+	localIP, err := consul.GetPrivateIP()
+	if err != nil {
+		return nil, err
+	}
+
+	client.clientAPI, err = api.NewClient(api.DefaultConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	srvRegInfo := &api.AgentServiceRegistration{
+		ID:      srvName,
+		Name:    srvName,
+		Address: localIP.String(),
+		Port:    svcPort,
+		Check: &api.AgentServiceCheck{
+			CheckID: srvName,
+			TTL:     checkTTL,
+		},
+	}
+
+	if err = client.Register(srvRegInfo); err != nil && client.IsReachable() {
```

Comment:
if the client is not Reachable, and we failed to register, we return `client, nil`. is the Client is valid state in that case?
this condition looks very suspicious

Reply:
This behavior is mentioned in a description of the function. If registration failed we can still use `Client` for register service again. If the client can't reach the local agent, we still can use a function to update TTL(yes, this function return error) and this behavior similar to the original Consul package.

Reply:
Then, what is the point of registration in `NewDefaultClient`?
I'm an important service, I  want to be registered. The return value won't tell me anything about it.
So, I must to call Register myself, or check, if I'm registered.
If registration isn't important for me, why calling it there?